src/topology/compacts.lean

```diff
@@ -234,4 +239,118 @@ let ⟨s, hs⟩ := exists_compact_subset is_open_univ $ mem_univ (classical.arbi
   ⟨{ carrier := s, compact' := hs.1, interior_nonempty' := ⟨_, hs.2.1⟩ }⟩
 
 end positive_compacts
+
+/-! ### Clopen sets -/
+
+/-- The type of clopen sets of a topological space. -/
+structure clopens (α : Type*) [topological_space α] :=
+(carrier : set α)
+(clopen' : is_clopen carrier)
+
+namespace clopens
+
+instance : set_like (clopens α) α :=
+{ coe := λ s, s.carrier,
+  coe_injective' := λ s t h, by { cases s, cases t, congr' } }
+
+lemma clopen (s : clopens α) : is_clopen (s : set α) := s.clopen'
+
+/-- Reinterpret a compact open as an open. -/
+@[simps] def to_opens (s : clopens α) : opens α := ⟨s, s.clopen.is_open⟩
+
+@[ext] protected lemma ext {s t : clopens α} (h : (s : set α) = t) : s = t := set_like.ext' h
+
+@[simp] lemma coe_mk (s : set α) (h) : (mk s h : set α) = s := rfl
+
+instance : has_sup (clopens α) := ⟨λ s t, ⟨s ∪ t, s.clopen.union t.clopen⟩⟩
+instance : has_inf (clopens α) := ⟨λ s t, ⟨s ∩ t, s.clopen.inter t.clopen⟩⟩
+instance : has_top (clopens α) := ⟨⟨⊤, is_clopen_univ⟩⟩
+instance : has_bot (clopens α) := ⟨⟨⊥, is_clopen_empty⟩⟩
+instance : has_sdiff (clopens α) := ⟨λ s t, ⟨s \ t, s.clopen.diff t.clopen⟩⟩
+instance : has_compl (clopens α) := ⟨λ s, ⟨sᶜ, s.clopen.compl⟩⟩
+
+instance : boolean_algebra (clopens α) :=
+set_like.coe_injective.boolean_algebra _ (λ _ _, rfl) (λ _ _, rfl) rfl rfl (λ _, rfl) (λ _ _, rfl)
+
+@[simp] lemma coe_sup (s t : clopens α) : (↑(s ⊔ t) : set α) = s ∪ t := rfl
+@[simp] lemma coe_inf (s t : clopens α) : (↑(s ⊓ t) : set α) = s ∩ t := rfl
+@[simp] lemma coe_top : (↑(⊤ : clopens α) : set α) = univ := rfl
+@[simp] lemma coe_bot : (↑(⊥ : clopens α) : set α) = ∅ := rfl
+@[simp] lemma coe_sdiff (s t : clopens α) : (↑(s \ t) : set α) = s \ t := rfl
+@[simp] lemma coe_compl (s : clopens α) : (↑sᶜ : set α) = sᶜ := rfl
+
+instance : inhabited (clopens α) := ⟨⊥⟩
+
+end clopens
+
+/-! ### Compact open sets -/
+
+/-- The type of compact open sets of a topological space. -/
```

Comment:
Maybe also mention spectral spaces here. As Yury points out, this is a somewhat surprising concept that is mostly useful in a non-Hausdorff context. But many people work with Hausdorff spaces 99% of the time.